pkg/metrics/metrics_test.go

```diff
@@ -613,6 +695,14 @@ func containsMetrics(expectedMfs, gotMfs []*cmg.MetricFamily) bool {
 						fmt.Println("difference in counter values")
 						continue
 					}
+
+				case got.Metric[i].Gauge != nil && expected.Metric[j].Gauge != nil:
```

Comment:
I do not see "expected" change?

Reply:
What do you mean by this? I just changed one of the test expectations from 51 to 52 and it failed as expected. Am I missing something here?

Reply:
I spent some time trying to consolidate verifyInFlightMetric into verifyMetrics, but I found some unexpected test framework behavior that made it difficult to integrate the two. `verifyInFlightMetric` works well for this specific test.

Reply:
The main issues were the following:
1. `verifyMetrics` silently passes when an expected metric is not formatted correctly
2. `sortMfs` returns an empty slice always, which causes test validation to pass incorrectly
3. ran into some nil pointers when getting `sortMfs` to work with two different metric types.

I can open an issue to update the existing `verifyMetrics`. Was hoping to get this PR in with `verifyInFlightMetrics` for now.